src/dmtest/gendatablocks.py

```diff
@@ -451,21 +448,29 @@ def write(self,
         if (compress < 0.0) or (compress > 0.96):
             raise ValueError("the compression fraction " + str(compress)
                              + " is invalid")
-        stream = BlockStream(tag, dedupe, compress, direct, sync, fsync)
+        if direct:
+            # Direct I/O requires special handling to ensure proper
+            # alignment of the in-memory buffer being written to the
+            # destination. We don't do that yet.
+            raise NotImplementedError("direct I/O is not yet supported")
+        stream = BlockStream(tag, dedupe, compress)
 
         flags = os.O_WRONLY
-        if stream.sync:
+        if sync:
             flags |= os.O_SYNC
         if self.create:
             flags |= os.O_CREAT
 
+        logging.info(f"writing {self.block_count*self.block_size} bytes tagged \"{tag}\""
+                     f" to {self.path} at {self.block_size*self.offset} open flags {flags}")
         with os.fdopen(os.open(self.path, flags), "r+b") as fd:
             self._seek(fd)
             for n in range(0, self.block_count):
                 data = stream.generate(n, self.block_size)
-                fd.write(data)
+                count = fd.write(data)
```

Comment:
Why does this need to be assigned to variable 'count'? I don't see count used anywhere else.

Reply:
Ah, good catch, thanks. I had added a check to see if we actually wrote the amount desired, but it didn't work, so removed (most of) it again...